main.go

```diff
@@ -281,6 +281,10 @@ func ReplaceInvalidChars(in *string) {
 			*in = (*in)[:charIndex] + "_" + (*in)[charIndex+1:]
 		}
 	}
+	// prefix with _ if first char is 0-9
+	if int((*in)[0]) >= 48 && int((*in)[0]) <= 57 {
+		*in = "_" + *in
```

Comment:
This doesn't seem right, it looks to be mutating a string - but strings are immutable in  Go.

Reply:
but the same string is also mutated at line 281 right? It does produce correct results locally. (`403_name -> _403_name etc`)

Reply:
Yeah, so I'm also wondering how that works and if it does if it's behaviour that can be depending on.

Reply:
take a look at this reddit topic, this explains it. 

https://www.reddit.com/r/golang/comments/57ica9/arent_strings_in_go_immutable/

Reply:
Thanks for the reference, seems it's okay though not code I'd use personally.

Reply:
Ok, no problem. If possible i would like to know how you would solve it, just trying to learn here.

Reply:
I'd take in a `string` and return a `string`. No need for any pointers.

Reply:
ah ok, but if you have a (non pointer) string as argument of your function it will copy the variable, which takes a bit of time and memory. Don't think that would be a problem, but still.